src/components/ui/dashboard/add-coaching-session-dialog.tsx

```diff
@@ -0,0 +1,129 @@
+  dialogTrigger,
+}: AddCoachingSessionDialogProps) {
+  const { currentCoachingRelationshipId } = useCoachingRelationshipStateStore(
+    (state) => state
+  );
+  const fromDate = DateTime.now().minus({ month: 1 });
+  const toDate = DateTime.now().plus({ month: 1 });
+  const { refresh } = useCoachingSessionList(
+    currentCoachingRelationshipId,
+    fromDate,
+    toDate
+  );
+  const { create: createCoachingSession } = useCoachingSessionMutation();
+  const [newSessionDate, setNewSessionDate] = useState<Date | undefined>(
+    undefined
+  );
+  const [newSessionTime, setNewSessionTime] = useState<string>("");
+  const { isCoach } = useAuthStore((state) => state);
+
+  const handleCreateSession = (e: React.FormEvent) => {
+    e.preventDefault();
+    if (!newSessionDate || !newSessionTime) return;
+
+    const [hours, minutes] = newSessionTime.split(":").map(Number);
+    const dateTime = getDateTimeFromString(newSessionDate.toISOString())
+      .set({ hour: hours, minute: minutes })
+      .toFormat("yyyy-MM-dd'T'HH:mm:ss");
+
+    const newCoachingSession: CoachingSession = {
+      ...defaultCoachingSession(),
+      coaching_relationship_id: currentCoachingRelationshipId,
+      date: dateTime,
+    };
+
+    createCoachingSession(newCoachingSession)
+      .then(() => {
+        refresh();
+        onCoachingSessionAdded();
+        setNewSessionDate(undefined);
+        setNewSessionTime("");
+      })
+      .catch((err: Error) => {
+        console.error("Failed to create new Coaching Session: " + err);
+        throw err;
+      });
+  };
+
+  return (
+    <Dialog open={open} onOpenChange={onOpenChange}>
+      <DialogTrigger asChild>
+        {React.cloneElement(dialogTrigger, {
+          ...(dialogTrigger.props as React.HTMLAttributes<HTMLButtonElement>),
+          className: cn(
+            (dialogTrigger.props as React.HTMLAttributes<HTMLButtonElement>)
+              .className,
+            isCoach ? "visible" : "invisible"
+          ),
+        })}
+      </DialogTrigger>
+
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle>Create New Coaching Session</DialogTitle>
+        </DialogHeader>
+        <form onSubmit={handleCreateSession} className="space-y-4">
+          <div className="space-y-2">
+            <Label htmlFor="session-date">Session Date</Label>
+            <Calendar
+              mode="single"
+              selected={newSessionDate}
+              onSelect={(date) => setNewSessionDate(date)}
+            />
+          </div>
+          <div className="space-y-2">
+            <Label htmlFor="session-time">Session Time</Label>
+            <input
+              type="time"
+              id="session-time"
+              value={newSessionTime}
+              onChange={(e) => setNewSessionTime(e.target.value)}
+              className="w-full border rounded p-2"
+              required
+            />
+          </div>
+          <Button type="submit">Create Session</Button>
```

Comment:
Similar to comment i left on Caleb's recent PR with a form, its good UX to disable the form submit button until all required fields are "dirty".

```suggestion
          <Button type="submit" disabled={!newSessionDate || !newSessionTime}>Create Session</Button>
```

Reply:
Thanks, great catch Zach.